### PR TITLE
Fix some memory leaks that affect routing and route creation

### PR DESCRIPTION
**Pull request by bitomaxsp**

### CODE REVIEW

**Comment by c9s** [comment]
Please make travis ci pass

File: include/r3_define.h [hunk 1]
```diff
@@ -9,9 +9,11 @@
 #define DEFINE_H
 #include <stdbool.h>
 
+
 #if !defined(bool) && !defined(__cplusplus)
 typedef unsigned char bool;
 #endif
+#endif
```
Comment (c9s) [inline include/r3_define.h:16]: This is duplicated

**Comment by c9s** [comment]
any update?

**Comment by bitomaxsp** [comment]
@c9s Can you reopen it, i'll try to finish it. 
This PR was ignored so long, that i gave up waiting response.

**Comment by c9s** [comment]
sure :) welcome back

**Comment by bitomaxsp** [comment]
All the fixes are already in. Closing this

**Pull request closed without merging**